implementations/rust/ockam/ockam_key_exchange_x3dh/src/initiator.rs

```diff
@@ -138,54 +154,22 @@ impl<V: X3dhVault> KeyExchanger for Initiator<V> {
                         .hkdf_sha256(&salt, CSUITE, Some(&ikm), vec![atts, atts])?;
                 let encrypt_key = keyrefs.pop().ok_or(X3DHError::InvalidState)?;
                 let decrypt_key = keyrefs.pop().ok_or(X3DHError::InvalidState)?;
-                let ek = self.vault.secret_public_key_get(&esk)?;
-                let pubkey = self.vault.secret_public_key_get(ephemeral_identity_key)?;
 
                 let mut state_hash = self.vault.sha256(CSUITE)?.to_vec();
                 state_hash.append(&mut ikm_bytes);
                 let state_hash = self.vault.sha256(state_hash.as_slice())?;
 
-                let mut aad = ek.as_ref().to_vec();
-                aad.extend_from_slice(&self.vault.sha256(pubkey.as_ref())?);
-                aad.extend_from_slice(CSUITE);
-                aad.extend_from_slice(&state_hash);
-
-                let atts = SecretAttributes::new(
-                    SecretType::Curve25519,
-                    SecretPersistence::Persistent,
-                    CURVE25519_SECRET_LENGTH,
-                );
-
-                let skb = if let Some(ik) = &self.identity_key {
-                    ik.clone()
-                } else {
-                    self.vault.secret_generate(atts)?
-                };
-                let ikb = self.vault.secret_public_key_get(&skb)?;
-
-                let mut plaintext = ikb.as_ref().to_vec();
```

Comment:
This doesn't belong to x3dh itself, and I'm not aware of why we had this here. Therefore, I'm removing everything that doesn't belong to x3dh. Pretty sure we'll revisit these choices soon.